kafka-rest/src/main/java/io/confluent/kafkarest/KafkaRestConfig.java

```diff
@@ -873,6 +873,14 @@ public Properties getProducerProperties() {
 
     Properties producerProperties = new Properties();
     producerProperties.putAll(producerConfigs);
+
+    // KREST-4606: Disable idempotency until KAFKA-13668 is fixed.
```

Comment:
Is this actually correct though? Would we allow people enabling idempotency without warning them that Kafka REST might not support it the way they expect it, even after KAFKA-13668 is fixed? If not (which is what I assume), let's remove or reword this comment.

Reply:
I agree - after our chat this morning, I think we came to the conclusion that idempotency in REST produce is a feature request.  While the REST produce client to kafka part of the flow might be idempotent, the http client to the REST produce client is not.

Reply:
Yes, we would. That's current behaviour. I think Kafka REST is very explicit that passing a `producer.` config is equivalent to passing down the config to the underlying producer. We are not talking about having a `idempotence.enable` config in Kafka REST itself. People should be aware that passing that config to the Producer changes the producer only, but has no other effect.

Reply:
I think this might easily violate the [principle of least astonishement](https://en.wikipedia.org/wiki/Principle_of_least_astonishment). People might e.g. assume that Kafka REST guarantees to reuse the same Producer underneath and that it somehow ensures idempotency guarantees across the whole write path. Without understanding the exact lifecycle of a Producer instance, which as we all know is not trivial, they can easily assume more than there is.
I think the added value to people that actually understand all the tricky details might be outweighed by the potential problems for people that don't, so I'd vote for not changing this even after KAFKA-13668 is fixed, and maybe adding an `force.enable.idempotency` Kafka REST config that can gate enabling the idempotency.

Reply:
If that was the case, I think we should disable idempotent producer for all versions, not just `7.0.x+`. I think that is out of scope for this PR. We should disable it now for `7.0.x+` because we have a real problem. And then we can have a discussion whether we just don't support this at all for all versions.

Reply:
Rephrased the comment to reflect our collective thinking.